astropy/modeling/core.py

```diff
@@ -2416,24 +2417,82 @@ def render_model(model, arr=None, coords=None, copy=True):
     ----------
     model : `Model`
         Model to be evaluated.
-    arr : `numpy.ndarray`, optional
-        Array on which the model is evaluated.
+    arr : `numpy.ndarray` 
+        Array on which the model is evaluated. 
     coords : array-like, optional
-        Coordinate arrays mapping to ``array``, such that 
+        Coordinate arrays mapping to ``arr``, such that 
         ``arr[coords] == arr``.
+    wcs : `~astropy.wcs.WCS`, optional
+        A WCS object mapping to the input ``arr``. 
+    origin: int, optional
+        The coordinate in the upper left corner of the image. In FITS and 
+        Fortran standards, this is 1. In Numpy and C standards this is 0. 
     copy : bool, optional
         Default is ``True``. If ``False``, ``arr`` will be modified in place. 
+    return_masked : bool, optional
+        If True, the output array is masked according to mask if provide, or 
+        outside of the bounding_box limits. The default is ``False``.
 
     Returns
     -------
-    array : `numpy.ndarray`
-        The model evaluated on the input ``arr`` or a new array from ``coords``.  
-        If ``arr`` and ``coords`` are both `None`, the returned array is 
-        limited to the `Model.bounding_box` limits. If 
-        `Model.bounding_box` is `None`, ``arr`` or ``coords`` must be passed.
+    arr : `numpy.ndarray`
+        The model evaluated on the input ``arr`` or ``hdu``, or a new array 
+        from ``coords``. If ``arr``, ``coords``, and ``hdu`` are all `None`, 
+        the returned array is limited to the `Model.bounding_box` limits. If 
+        `Model.bounding_box` is `None`; ``arr``, ``hdu``, or ``coords`` must 
+        be given.
+
+    Examples
+    --------
+    .. plot::
```

Comment:
Maybe it's best to keep the examples in the high-level docs?
Using world coordinates for model parameters is a new concept and needs an extra section and example.
Maybe it's possible to re-use parts for the pixel-coordinate based example via utility functions.
Maybe compute the exact same example image, once with pixel and once with world coordinates?